src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/PerformanceTrackerService.cs

```diff
@@ -58,24 +57,32 @@ internal PerformanceTrackerService(double minLOFValue, double averageThreshold,
             _stddevThreshold = stddevThreshold;
 
             _gate = new object();
-            _queue = new PerformanceQueue(SampleSize);
+
+            // We require at least 100 samples for background document analysis result to be stable
+            _queueForDocumentAnalysis = new PerformanceQueue(maxSampleSize: 300, minSampleSize: 100);
+
+            // We require at least 10 samples for span/lightbulb analysis result to be stable
+            _queueForSpanAnalysis = new PerformanceQueue(maxSampleSize: 30, minSampleSize: 10);
```

Comment:
@jmarolf @CyrusNajmabadi - I am not sure what is a good sample size to use to report this outlier analyzer telemetry for lightbulb code path. For background document analysis, the performance reporting was set to max 300 samples and min of 100 samples for stable data. However, that seems too big a sample size for Ctrl + Dot execution, but I am not sure if 30/10 is a good sample size. Any suggestions here?

Reply:
I don't see why we think these needs to be smaller. Do we believe that there are not that many span-based invocations in a session?

Reply:
I am assuming there are more invocations of background analysis of open files then lightbulb sessions, hence the different sample sizes. Though I agree that 100 samples (which would mean around 25 lightbulb invocations) would easily be exceeded. I can change this min sample size as we deem appropriate.